src/Units/MoveTypes/MoveType.java

```diff
@@ -68,9 +69,27 @@ public int getMoveCost(Weathers weather, TerrainType terrain)
     return cost.intValue();
   }
 
-  public FloodFillFunctor getUnitMoveFunctor(Unit mover, boolean includeOccupied, boolean canTravelThroughEnemies)
+  public int getTransitionCost(GameMap map, XYCoord from, XYCoord to)
   {
-    return new BasicMoveFillFunctor(mover, this, includeOccupied, canTravelThroughEnemies);
+    return getTransitionCost(map, from, to, null, true);
+  }
+  public int getTransitionCost(GameMap map, XYCoord from, XYCoord to,
+                               Unit mover, boolean canTravelThroughEnemies)
+  {
+    // if we're past the edges of the map
+    if( !map.isLocationValid(to) )
+      return MoveType.IMPASSABLE;
+
+    // note to self: extend this if we ever support moving to non-adjacent tiles
+    int cost = getMoveCost(map.getEnvironment(to.xCoord, to.yCoord));
+
+    // if there is an enemy unit in that space
+    if( !canTravelThroughEnemies
+        && (map.getLocation(to).getResident() != null)
+        && mover.CO.isEnemy(map.getLocation(to).getResident().CO) )
+      cost = MoveType.IMPASSABLE;
```

Comment:
This code will currently blow up IFF there are units in the way, you pass in a `null` unit, and `canTravelThroughEnemies` is unset.
Should a `null` unit input simply count *everyone* as enemies?
Is the above just an error condition we should avoid?

I've been sort of treating it as a malformed call, but IDK if that's the right idea since that seems like something that *could* blow up at some point.
Maybe the argument should be an allegiance thing, so you input your team number and there's no option for `null`?

Reply:
Are there any cases where a `null` unit could be passed in? Though it might make more sense to pass in a `Commander` instead, since we only use it for a friendliness check.

I think it's fair to either treat `null` as an exception (in which case, raise something) or by treating everyone as enemies.

Reply:
Yes. AI might (will) want to calculate paths without having an actual unit yet. That said, it's also easy to make an ephemeral `Unit` for the call.
I think I agree; passing in an `Army` makes it intuitive that a `null` says you're enemies with everyone.